packages/core/src/errors.ts

```diff
@@ -48,3 +49,9 @@ export class InvalidStringError extends CustomError {
     super(`Invalid string: "${expectation}"`);
   }
 }
+
+export class InvalidNativeScript extends CustomError {
```

Comment:
This is really a variant of SerializationError since it's thrown when mapping to CSL. It's a common error type in coreToCsl, so I think we want to `InvalidNativeScript instanceof SerializationError === true`

Although this error might also be useful in a different context in the future (like mapping from cardano-cli json), so not 100% sure about this one.

```suggestion
export class InvalidNativeScript extends SerializationError {
```

Reply:
I will remove it and use the SerializationError instead, I think we can add it back later down the line if needed.

Reply:
addressed here https://github.com/input-output-hk/cardano-js-sdk/pull/377/commits/b5a80d971f77d88933f07c51622f45ec4028b7b4